src/Microsoft.AspNetCore.WebUtilities/HttpRequestStreamReader.cs

```diff
@@ -105,25 +85,23 @@ public HttpRequestStreamReader(Stream stream, Encoding encoding, int bufferSize)
                     _charPool.Return(_charBuffer);
                     _charBuffer = null;
                 }
+                throw;
             }
         }
 
         protected override void Dispose(bool disposing)
         {
-            if (disposing && _stream != null)
+            if (disposing && !_disposed)
             {
-                _stream = null;
-
+                _disposed = true;
                 if (_bytePool != null)
```

Comment:
why are there null checks on _bytePool and _charPool here? The constructor prevents nulls.